libraries/AP_Scripting/examples/test_load.lua

```diff
@@ -0,0 +1,19 @@
+--[[
+ test the load function for loading new code from strings
+--]]
+
+gcs:send_text(0,"Testing load() method")
+
+local str = [[
+function(x,y)
+  return math.sin(x) + math.cos(y)
+end
+]]
+
+local f,errloc,err = load("return " .. str,"TestFunc","t",_ENV)
+if not f then
+   gcs:send_text(0,string.format("Error %s: %s", errloc, err))
+else
+   _ENV["TestFunc"] = f()
```

Comment:
```suggestion
   TestFunc = f()
```
This seems to work, and is less confusing.

You can also give then name in the loaded string with:
```
local str = [[
function TestFunc(x,y)
  return math.sin(x) + math.cos(y)
end
]]
local f, errloc, err = load(str,"TestFunct","t",_ENV)
f()
TestFunc(3,4)
```
It removes the need for the pre-pending return.

Reply:
thanks! I've updated the example, and also made it use a pcall() for running the loaded function, which I think would be good practice